src/com/facebook/buck/event/listener/TestResultFormatter.java

```diff
@@ -218,11 +218,15 @@ public void reportResultSummary(
       }
     }
 
-    if (summaryVerbosity.getIncludeStdOut() && testResult.getStdOut() != null) {
+    if (summaryVerbosity.getIncludeStdOut()
+        && testResult.getStdOut() != null
+        && testResult.getStdOut().length() > 0) {
       addTo.add("====STANDARD OUT====", testResult.getStdOut());
     }
 
-    if (summaryVerbosity.getIncludeStdErr() && testResult.getStdErr() != null) {
+    if (summaryVerbosity.getIncludeStdErr()
+        && testResult.getStdErr() != null
```

Comment:
you can simplify by using [Strings.isNullOrEmpty](https://google.github.io/guava/releases/23.0/api/docs/com/google/common/base/Strings.html#isNullOrEmpty-java.lang.String-)